src/components/FloatingFlow.vue

```diff
@@ -214,10 +221,31 @@ export default {
       this.bringDialogToFront(id);
     },
     resetApp: function(){
-      Object.assign(this.$data, initialState());
-      var closeItems = document.querySelectorAll('.mapboxgl-popup-close-button');
+      this.setState(initialState());
+    },
+    setState: function(state){
+      this.mainTabName = state.mainTabName;
+      this.zIndex = state.zIndex;
+      this.showDialogIcons = state.showDialogIcons;
+      this.dockedArray = [];
+      Object.assign(this.dockedArray, state.dockedArray);
+      this.activeDockedId = state.activeDockedId;
+      this.currentCount = state.currentCount;
+      this.entries = [];
+      Object.assign(this.entries, state.entries);
+      let closeItems = document.querySelectorAll('.mapboxgl-popup-close-button');
       closeItems.forEach( (item) => { item.click() });
     },
+    getState: function() {
+      let state = JSON.parse(JSON.stringify(this.$data));
```

Comment:
is the purpose of this to break the refence to `this.$data`?

Could use `let state = ...this.$data` if so.

Reply:
No, it is to copy without any of the vue reactive properties.